include/seqan3/core/algorithm/concept.hpp

```diff
@@ -0,0 +1,149 @@
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides concepts for the configuration classes.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <functional>
+#include <type_traits>
+
+#include <meta/meta.hpp>
+
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/std/concept/callable.hpp>
+#include <seqan3/std/concept/core_language.hpp>
+#include <seqan3/std/concept/object.hpp>
+
+namespace seqan3::detail
+{
+
+// ----------------------------------------------------------------------------
+// Concept config_element_concept
+// ----------------------------------------------------------------------------
+
+//!\cond
+template <typename derived_t>
+class config_element_base;
+//!\endcond
+
+/*!\interface seqan3::detail::config_element_concept <>
+ * \brief Concept for an algorithm configuration.
+ * \ingroup algorithm
+ *
+ * \extends seqan3::semi_regular_concept
+ * \extends seqan3::assignable_concept
+ */
+
+/*!\name Requirements for seqan3::detail::config_element_concept
+ * \relates seqan3::detail::config_element_concept
+ * \brief   You can expect this member on all types that satisfy seqan3::detail::config_element_concept.
+ * \{
+ */
+/*!\var value
+ * \memberof seqan3::detail::config_element_concept
+ * \brief Member storing the configuration value.
+ */
+//!\}
+//!\cond
+template <typename config_t>
+concept bool config_element_concept = semi_regular_concept<std::remove_reference_t<config_t>> &&
+                                      assignable_concept<std::remove_reference_t<config_t> &, config_t> &&
```

Comment:
I think assignable_concept with one parameter already does what you want, or not?